spec/index.bs

```diff
@@ -1069,11 +1275,11 @@ interface GPUAdapter {
                                 is a subset of those in
                                 |adapter|.{{adapter/[[extensions]]}}.
 
-                            - For each type of limit in {{GPULimits}},
-                                the value of that limit in
-                                |descriptor|.{{GPUDeviceDescriptor/limits}}
-                                is no [=better=] than the value of that limit in
-                                |adapter|.{{adapter/[[limits]]}}.
+                            - For each member |limitName| of {{GPULimitName}},
+                                the value of
+                                |descriptor|.{{GPUDeviceDescriptor/limits}}[|limitName|]
+                                is no [=limit/better=] than the value of
+                                |adapter|.{{adapter/[[limits]]}}[|limitName|].
```

Comment:
How can the user guarantee that? they don't have access to `|adapter|.{{adapter/[[limits]]}}`.

Is it valid for the user to just request some limits at device creation without checking the adapter for them?

Reply:
It would be up to them to use supportsLimit or getLimit to determine the supported limits before requesting higher-than-default ones.

Reply:
I think there is an opportunity here to do better, and encourage more portable programming. I.e., with this PR the user has 2 different ways to check if a custom limit is ok: via the adapter methods, or simply by requesting the device.

I wonder if it should say that: if a limit hasn't been exposed by the adapter method, the user agent will always consider it to be the baseline limit. Of course, we can do it in Firefox, anyway, and it will be ok for the spec.

Reply:
This would make the adapter stateful (tracking the highest value returned for each limit so far), right? I'd be hesitant because if you have multiple WebGPU apps on the same page, they'd get the same adapters and share the same internal state. Though that sharing has fairly minimal impact on how the api behaves.

Instead, should requestAdapter take a dictionary of "highest limits I might be interested in", and all limits on the GPUAdapter would be under those (and exposed directly as before, instead of with this query)? Downside here is that one hardware adapter could map to many GPUAdapter objects (depending on the limits of the request). But maybe that's fine.

Reply:
Or as I just thought of on #1098, maybe it should just be a browser warning instead of a spec error.

Reply:
The whole point of the privacy budget (and your PR) is to make the adapter stateful. The budget is its state.

Reply:
That state doesn't have to be immediately observable to the application though. In this case it leaks between independent users of WebGPU on the same page causing small behavior changes.

Reply:
Is this resolved?

Reply:
I don't think so? My point is, there is no exact value of `adapter/[[limits]]`. There is no limit until you query for it. We could specify this as the `[[limits]]` internal slot is populated as you are making queries, but even in this case there needs to be some wording/algorithm here branching on whether the `[[limits]]` contain a particular limit or not.

Reply:
We don't have to be in the business of specifying exactly how `[[limits]]` changes due to privacy budgeting. I think we should only try to make sure there is measurable API surface, and not try to specify the spending of privacy budget. If an implementation wants to implement privacy budgeting it can effectively fill out the `[[limits]]` slot lazily, within the as-if nature of the specification.

Reply:
I see. I guess my disagreement is that I don't think "as-if" rule would help here. We need to talk! Maybe on an extra call, or matrix